Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +379,124 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
 
 PTF_TEST_CASE(TestPcapLiveDeviceBlockingMode)
 {
-	// open device
-	pcpp::PcapLiveDevice* liveDev = pcpp::PcapLiveDeviceList::getInstance().getPcapLiveDeviceByIp(PcapTestGlobalArgs.ipToSendReceivePackets.c_str());
-	PTF_ASSERT_NOT_NULL(liveDev);
-	PTF_ASSERT_TRUE(liveDev->open());
-	DeviceTeardown devTeardown(liveDev);
+	std::vector<pcpp::PcapLiveDevice::DeviceConfiguration> configs;
+	configs.emplace_back(); // the default config
 
-	// sanity - test blocking mode returns with timeout
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 5), -1);
+#if !defined(_WIN32)
+	configs.emplace_back(); // the config used poll
+	configs[1].usePoll = true;
+#endif
 
-	// sanity - test blocking mode returns before timeout
-	int packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeNoTimeout, &packetCount, 30), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+	// test the common behaviour for all configs
+	for(const auto & config: configs)
```

Comment:
by doing this, we can test many different configs.

Reply:
Do you plan to add more configs to this list?

Reply:
correctly no, but I think it's better than duplicating the code twice.